fn.go

```diff
@@ -197,8 +208,32 @@ func (f *Function) transformK8sResource(input *v1beta1.Input) composed.Unstructu
 	return unstructuredData
 }
 
-// Check if user is not a bot
-func realUsers(members []string, api *slack.Client) []string {
+// Get initial resources
+func getInitialResources(req *fnv1beta1.RunFunctionRequest) (*initialResources, error) {
+	input := &v1beta1.Input{}
+	if err := request.GetInput(req, input); err != nil {
+		fmt.Println(err)
+	}
+	desired, err := request.GetDesiredComposedResources(req)
+	if err != nil {
+		return nil, err
+	}
+	api := slack.New(input.SlackAPIToken)
+	rsp := response.To(req, response.DefaultTTL)
+
+	xr, err := request.GetObservedCompositeResource(req)
+	if err != nil {
+		return nil, err
+	}
+	return &initialResources{Input: input, API: api, XR: xr, Rsp: rsp, Desired: desired, CurrentTimestamp: int(time.Now().Unix())}, nil
+}
+
+// Get and process slack members
+func processSlackMembers(api *slack.Client, channelID string) ([]string, error) {
```

Comment:
@ca7alindev let's move slack related parts into separate package, because can be another channel of polling in the future.